api/v1/controllers/samples.js

```diff
@@ -159,25 +159,46 @@ module.exports = {
    * @param {Function} next - The next middleware function in the stack
    */
   upsertSample(req, res, next) {
-    const resultObj = { reqStartTime: new Date() };
-    u.getUserNameFromToken(req,
-      featureToggles.isFeatureEnabled('enforceWritePermission'))
-    .then((userName) =>
-      helper.model.upsertByName(req.swagger.params.queryBody.value, userName)
-    )
-    .then((o) => {
-      resultObj.dbTime = new Date() - resultObj.reqStartTime;
+    if (featureToggles.isFeatureEnabled(constants.featureName)) {
+      const resultObj = { reqStartTime: new Date() }; // for logging
+      const sampleQueryBody = req.swagger.params.queryBody.value;
 
-      // loop through remove values to delete property
-      if (helper.fieldsToExclude) {
-        u.removeFieldsFromResponse(helper.fieldsToExclude, o.dataValues);
-      }
+      redisModelSample.upsertSampleInRedis(
+        sampleQueryBody, resultObj, res.method
+      )
+      .then((sampleRes) => {
+        resultObj.dbTime = new Date() - resultObj.reqStartTime;
```

Comment:
lines 170-180 and lines 189-200 are almost the same... I don't want you to spend more than a couple of minutes thinking about this, but if it's easy to do, can the duplicate-ish code be merged?